src/Lykke.Service.BlockchainWallets/Workflow/CommandHandlers/EndBalanceMonitoringCommandHandler.cs

```diff
@@ -31,23 +36,33 @@ public class EndBalanceMonitoringCommandHandler
 
             try
             {
-                var apiClient = _blockchainIntegrationService.TryGetApiClient(command.IntegrationLayerId);
+                const MonitoringSubscriptionType subscriptionType = MonitoringSubscriptionType.Balance;
+
+                var address = command.Address;
+                var assetId = command.AssetId;
+                var blockchainType = command.BlockchainType;
+
+                var apiClient = _blockchainIntegrationService.TryGetApiClient(blockchainType);
 
                 if (apiClient != null)
                 {
-                    await apiClient.StopBalanceObservationAsync(command.Address);
-
-                    publisher.PublishEvent(new WalletDeletedEvent
+                    if (!await _monitoringSubscriptionRepository.AddressIsSubscribedAsync(blockchainType, address, subscriptionType))
```

Comment:
Why do not we do ```apiClient.StopBalanceObservationAsync``` if ```_monitoringSubscriptionRepository.AddressIsSubscribedAsync```? Looks like condition should be inverted. And, then ```UnregisterWalletSubscriptionAsync``` could be moved under this condition too.